include/libdnf/repo/package_downloader.hpp

```diff
@@ -53,9 +53,17 @@ class PackageDownloader {
     void add(const libdnf::rpm::Package & package, const std::string & destination, void * user_data = nullptr);
 
     /// Download the previously added packages.
-    /// @param fail_fast Whether to fail the whole download on a first error or keep downloading.
-    /// @param resume Whether to try to resume the download if a destination package already exists.
-    void download(bool fail_fast, bool resume);
+    void download();
+
+    /// Configure whether to fail the whole download on a first error or keep downloading.
+    /// @param value If true, download will fail on the first error, otherwise it continues.
+    /// This is set to true by default.
+    void set_fail_fast(bool value);
+
+    /// Configure whether to try resuming the download if a destination package already exists.
+    /// @param value If true, download is tried to be resumed, otherwise it starts over.
+    /// This is set to true by default.
+    void set_resume(bool value);
```

Comment:
In this approach the behavior will be not modifiable from DNF5 manager. Only API user can make a decision what approach should be used. Could you imagine a situation when resume would create problem?

Reply:
If we would provide configuration options for `resume` or `fail_fast` then I understand it like these values would have global meaning. But we are already using specific values for different scenarios inside the libdnf. So the question is, how it will work then? Should the user configuration override the values we currently pass to the downloader in the library?

Reply:
make sense, I overlooked the internal modification for download pgpkeys. it makes no sense to resume interrupted downloads for very small files.